mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsTypes.h

```diff
@@ -439,6 +465,23 @@ class AccessorImplDeviceType
   llvm::ArrayRef<mlir::Type> getBody() const;
 };
 
+class AccessorSubscriptType
+    : public Type::TypeBase<AccessorSubscriptType, Type,
+                            detail::AccessorSubscriptStorage,
+                            mlir::MemRefElementTypeInterface::Trait,
+                            mlir::LLVM::PointerElementTypeInterface::Trait> {
+public:
+  using Base::Base;
+
+  static mlir::sycl::AccessorSubscriptType
+  get(MLIRContext *Context, unsigned int CurrentDimension,
+      llvm::SmallVector<mlir::Type, 4> Body);
+  static mlir::Type parseType(mlir::DialectAsmParser &Parser);
+
```

Comment:
Should we have an additional `mlir::AccessorCommonType` member here hosting the type this class is a member of? This would allow to convert to an LLVM type of the same name as the one created by clang.

Reply:
I do not thing that is strictly necessary because AFAIK the accessor_common class does not contain member variables of its own.